layouts/partials/menu.html

```diff
@@ -16,8 +16,20 @@
   <div class="highlightable">
     <ul class="topics">
       {{ $page := . }}
-      {{ range $key , $value := .Site.Sections }}
+      {{ $.Scratch.Set "menuItems" .Site.Params.menu }}
+      {{ if not .Site.Params.menu }}
+        {{ $.Scratch.Set "menuItems" .Site.Sections }}
+      {{ end }}
+      {{ $menuItems := $.Scratch.Get "menuItems" }}
+      {{ range $i, $key := $menuItems }}
       {{ if ne $key "" }}
+
+      {{ if not $.Site.Params.menu }}
+        {{ $.Scratch.Set "currentItem" (index $page.Site.Sections $i) }}
+      {{ else }}
+        {{ $.Scratch.Set "currentItem" (index $page.Site.Sections $key) }}
```

Comment:
I changed it to show warning about possible typos in `config.toml`. I thought it would be better than skipping it silently.